src/lumigo_tracer/utils.py

```diff
@@ -161,3 +162,62 @@ def is_aws_environment():
     :return: heuristically determine rather we're running on an aws environment.
     """
     return bool(os.environ.get("LAMBDA_RUNTIME_DIR"))
+
+
+class Frame:
+    MAX_VAR_LEN = 200
+
+    total_frames_size: int = 0
+
+    def __init__(self, frame_info: inspect.FrameInfo):
+        self.lineno: int = frame_info.lineno
+        self.file_name: str = frame_info.filename
+        self.function: str = frame_info.function
+        Frame.total_frames_size += len(str(self.lineno)) + len(self.file_name) + len(self.function)
+        self.variables: FrameVariables = Frame._truncate_locals(frame_info.frame.f_locals)
+
+    def to_dict(self) -> Dict[str, Union[int, str, FrameVariables]]:
+        return {
+            "lineno": self.lineno,
+            "file_name": self.file_name,
+            "function": self.function,
+            "variables": self.variables,
+        }
+
+    @staticmethod
+    def _truncate_var(var: Any) -> str:
+        var = str(var)
+        if len(var) > Frame.MAX_VAR_LEN:
+            return var[: Frame.MAX_VAR_LEN] + "..."
+        return var
+
+    @staticmethod
+    def _truncate_locals(f_locals: Dict[str, Any]) -> FrameVariables:
+        """
+        Truncate variable part or the entire variable in order to avoid exceeding the maximum frames size.
+        :param f_locals: inspect.FrameInfo.frame.f_locals
+        """
+        locals_truncated: FrameVariables = {}
+        for var_name, var_value in f_locals.items():
+            if not callable(var_value):
+                var = {var_name: Frame._truncate_var(var_value)}
+                var_len = len(json.dumps(var))
+                if var_len + Frame.total_frames_size > MAX_VARS_SIZE:
+                    return locals_truncated
+                Frame.total_frames_size += var_len
+                locals_truncated.update(var)
+        return locals_truncated
+
+
+def extract_frames_from_exception() -> List[Frame]:
+    """
+    Should be called only in `except` scope.
+    """
+    Frame.total_frames_size = 0
+    frames: List[Frame] = []
+    frames_infos = inspect.trace()
+    for frame_info in frames_infos[::-1]:
+        if Frame.total_frames_size >= MAX_VARS_SIZE or "lumigo_tracer" in frame_info.filename:
+            return frames
+        frames.append(Frame(frame_info))
```

Comment:
you are reverse the order of the frames, which is not good in the case of truncated information - 
the most recent frames should be first

Reply:
Ummm... I'm not sure that I'm right...
Check it :)

Reply:
I reverse the order of frames exactly for this purpose :)